pulp_container/app/registry_api.py

```diff
@@ -475,6 +479,161 @@ def get_queryset(self, *args, **kwargs):
         return (public_repositories | accessible_repositories).distinct()
 
 
+class FlatpakIndexDynamicView(APIView):
+    """
+    Handles requests to the /index/dynamic endpoint
+    """
+
+    def get(self, request):
+        unsatisfiable = False
+        req_repository = None
+        req_tags = set()
+        req_os = None
+        req_architecture = None
+        req_label_exists = {"org.flatpak.ref"}
+        req_label_values = {}
+        for key, values in request.query_params.lists():
+            if key == "repository":
+                req_repository = values[0]
+                if any(v != req_repository for v in values[1:]):
+                    unsatisfiable = True
+                    break
+            elif key == "tag":
+                req_tags.update(values)
+            elif key == "os":
+                req_os = values[0]
+                if any(v != req_os for v in values[1:]):
+                    unsatisfiable = True
+                    break
+            elif key == "architecture":
+                req_architecture = values[0]
```

Comment:
So with the updated understanding about multiple `architecture=...` query parameters, my understanding here would be that: if no `architecture=...` query parameters are given then all images are returned unfiltered; otherwise, all images are returned that match one of the given `architecture=...` query parameters.  (As is now implemented by <https://github.com/pulp/pulp_container/pull/1301/commits/719a9f599ba647de371a21450ab45f8c2af52366> "Fix logic for repeated query parameters".)

@owtaylor ?

Reply:
yes, that would be my understanding of OR logic in the repeated query params